packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -308,90 +386,163 @@ export class IamPolicyChecksWebview extends VueWebview {
                         cfnParameterPath === ''
                             ? `cfn-policy-validator check-access-not-granted --template-path ${document} --region ${this.region} --actions ${actions}`
                             : `cfn-policy-validator check-access-not-granted --template-path ${document} --region ${this.region} --actions ${actions} --template-copnfiguration-file ${cfnParameterPath}`
-                    this.executeCustomPolicyChecksCommand(cfnCommand)
+                    this.executeCustomPolicyChecksCommand(
+                        cfnCommand,
+                        cfnParameterPath ? true : false,
+                        documentType,
+                        'CheckAccessNotGranted'
+                    )
                     return
                 } else {
-                    this.onCustomPolicyCheckResponse.fire([IamPolicyChecksConstants.IncorrectFileExtension, 'red'])
+                    this.onCustomPolicyCheckResponse.fire([
+                        IamPolicyChecksConstants.IncorrectFileExtension,
+                        getResultCssColor('Error'),
+                    ])
                     return
                 }
             }
         }
     }
 
-    public executeValidatePolicyCommand(command: string) {
-        try {
-            this.handleValidatePolicyCliResponse(execSync(command).toString())
-        } catch (err: any) {
-            this.onValidatePolicyResponse.fire([parseCliErrorMessage(err.message), 'red'])
-        }
+    public executeValidatePolicyCommand(
+        command: string,
+        cfnParameterPathExists: boolean,
+        documentType: PolicyChecksDocumentType,
+        policyType?: PolicyChecksPolicyType
+    ) {
+        telemetry.accessanalyzer_iamPolicyChecksValidatePolicy.run(span => {
+            try {
+                span.record({
+                    cfnParameterFileUsed: cfnParameterPathExists,
+                    documentType: documentType,
+                    inputPolicyType: policyType ? policyType : 'None',
+                })
+                const resp = execSync(command)
+                const findingsCount = this.handleValidatePolicyCliResponse(resp.toString())
+                span.record({
+                    findingsCount: findingsCount,
+                })
+            } catch (err: any) {
+                if (err.status === 2) {
+                    //CLI responds with a status code of 2 when findings are discovered
+                    const findingsCount = this.handleValidatePolicyCliResponse(err.stdout.toString())
+                    span.record({
+                        findingsCount: findingsCount,
+                    })
+                } else {
+                    span.record({
+                        findingsCount: 0,
+                    })
```

Comment:
nit: you can add this to your first record on line 415. Fields can be re-recorded as much as you want, and the latest recording will be the one that makes it into the metric.